cmds/tail/tail.go

```diff
@@ -0,0 +1,261 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Tail prints the lasts 10 lines of a file. Can additionally follow the
+// the end of the file as it grows.
+//
+// Synopsis:
+//     tail [-f] [-n lines_to_show] [FILE]
+//
+// Description:
+//     If no files are specified, read from stdin.
+//
+// Options:
+//     -f: follow the end of the file as it grows
+//     -n: specify the number of lines to show (default: 10)
+
+// Missing features:
+// - follow-mode (i.e. tail -f)
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"syscall"
+)
+
+var (
+	flagFollow   = flag.Bool("f", false, "follow the end of the file")
+	flagNumLines = flag.Int("n", 10, "specify the number of lines to show")
+)
+
+type TailConfig struct {
+	follow   bool
+	numLines int
+}
+
+// getBlockSize returns the block size for the disk where the file pointed by
+// the reader resides. The block size is architecture-dependent, so I am using
+// `int64` as return type, since it is a superset of all the supported Blksize
+// sizes.
+// This is acceptable because this function is not used to get the exact block
+// size, but a value that is good enough to minimize the number of `read` syscalls
+func getBlockSize(f os.File) (int64, error) {
+	var stat syscall.Stat_t
+	err := syscall.Stat(f.Name(), &stat)
+	if err != nil {
+		return 0, err
+	}
+	return int64(stat.Blksize), nil
+}
+
+// readLastLinesBackwards reads the last N lines from the provided file, reading
+// backwards from the end of the file. This is more efficient than reading from
+// the beginning, but can only be done on seekable files, (e.g. this won't work
+// on stdin). For non-seekable files see readLastLinesFromBeginning.
+// It returns an error, if any. If no error is encountered, the File object's
+// offset is positioned after the last read location.
+func readLastLinesBackwards(inFile os.File, writer io.Writer, numLines int) error {
```

Comment:
done

Reply:
Shall I just let the function error out if the passed `os.File` is `nil`? (i.e. if `inFile` is `nil`, `Seek` will return an error but won't `panic()`). Otherwise I can add a check, which would be duplicated in the two functions.